src/java.base/share/classes/java/net/DatagramSocket.java

```diff
@@ -946,7 +947,18 @@ public DatagramChannel getChannel() {
      * @see       java.net.DatagramSocketImplFactory#createDatagramSocketImpl()
      * @see       SecurityManager#checkSetFactory
      * @since 1.3
+     *
+     * @deprecated Use {@link DatagramChannel}, or subclass {@code DatagramSocket}
+     *    directly.
+     *    <br> This method provided a way in early JDK releases to replace the
+     *    system wide implementation of {@code DatagramSocket}. It has been mostly
+     *    obsolete since Java 1.4. If required, a {@code DatagramSocket} can be
+     *    created to use a custom implementation by extending {@code DatagramSocket}
+     *    and using the {@linkplain #DatagramSocket(DatagramSocketImpl) protected
+     *    constructor} that takes an {@linkplain DatagramSocketImpl implementation}
+     *    as a parameter.
```

Comment:
This wording is good, are you planning to add this to the deprecated methods in Server and ServerSocket too?

Reply:
Can do -- see commit e36f91d. I've added similar wording to the equivalent methods in Server and SeverSocket